### ISSUE TITLE
Renaming layers destroys plot

### ISSUE THREAD

**Issue by AljoSt** [keras-team/keras#12046]
Please make sure that the boxes below are checked before you submit your issue.
If your issue is an **implementation question**, please ask your question on [StackOverflow](http://stackoverflow.com/questions/tagged/keras) or [on the Keras Slack channel](https://keras-slack-autojoin.herokuapp.com/) instead of opening a GitHub issue.

Thank you!

- [x] Check that you are up-to-date with the master branch of Keras. You can update with:
`pip install git+git://github.com/keras-team/keras.git --upgrade --no-deps`

- [x] Check that your version of TensorFlow is up-to-date. The installation instructions can be found [here](https://www.tensorflow.org/get_started/os_setup).

- [x] Provide a link to a GitHub Gist of a Python script that can reproduce your issue (or just copy the script here if it is short).

Hello,

When I create a model and then rename its layers (by setting the`.name` attribute) and then plot it, the plot becomes corrupted (the layers are plotted side by side without any connection). I guess this is due to the reason, that Keras relies on the names of layers to connect them and renaming breaks this(?).
Is there any way, to rename my layers without messing up the model? 
Saving the model summary as a `.txt` file seems to work btw.

Here is some code 
```
    input = Input(shape=(128, 128, 3))
    x = input
    for i in range(10):
        x = Conv2D(10, kernel_size=(3,3), name="layer{}".format(i))(x)
    model = Model(input, outputs=x, name="decoder")
    model.compile(optimizer='nadam', loss='categorical_crossentropy', metrics=['accuracy'])

    from keras.utils import plot_model

    outpath = "./test1.png"
    plot_model(model, show_shapes=True, to_file=outpath)
    for layer in model.layers:
        layer.name = "1" + layer.name

    outpath = "./test2.png"

    plot_model(model, show_shapes=True, to_file=outpath)
```

Edit:
Looking into the plot function (`keras.utils.vis_utils`) shows that this is due to the fact that `model._network_nodes` is not updated

**Comment by msymp** [comment]
Hello @AljoSt , here is one post from Stack Overflow and one previous GitHub Keras issue, which you could study, which on the similar lines as your query:

- https://stackoverflow.com/questions/49550182/keras-rename-model-and-layers
- https://github.com/keras-team/keras/issues/6194

Please let us know. Thanks.

**Comment by AljoSt** [comment]
Thanks for the quick response. 
I saw both threads before, but don't think they help me. I am able to change the names of the layers by changing `layer.name`, however, I can't plot the model afterwards anymore (or rather there are edges missing in the resulting plot). I don't know if renaming the layers destroys my model completely or if it just affects the `plot` function...

**Comment by jvishnuvardhan** [comment]
@AljoSt Is this still an issue? If yes, Please fill the issue [template](https://github.com/keras-team/keras/issues/new?template=a--tensorflow-backend-users.md). Could you update them if they are relevant in your case, or leave them as N/A? Along with the template, please provide as many details as possible to find the root cause of the issue. It would be great if you can provide a small code to reproduce the error. Thanks!

**Comment by jvishnuvardhan** [comment]
Closing due to lack of recent activity. Please update the issue when new information becomes available, and we will open a new issue. Thanks!

**Comment by was84san** [comment]
The performance when I changed the layers name is totally destroyed. Like I have three models with accuracy around 71, 70, 69. and when I tried to ensemble the models I got accuracy 5 % because i have to change the layers name for each model. Is this related with naming the layers

**Comment by jvishnuvardhan** [comment]
@was84san Please open a new issue with details of your issue and context. Also, fill the issue template. Thanks!

**Comment by aningineer** [comment]
Any solutions to this yet?
